### ISSUE TITLE
CallbackController.cs line 85 - variable "tempState" is never used???

### ISSUE THREAD

**Issue by AugustZellmer** [IntuitDeveloper/Oauth2-MVC5-DotnetSampleApp#4]
The code goes through all of this work to create a GUID, store it, and pass it to the auth endpoint, but (unless I'm misunderstanding something) it never actually checks to verify that the auth endpoint returns the correct GUID.

**Comment by nimisha84** [comment]
You are right. It should be checked on the client side as per docs to make sure that state returned is the same after callback. if you are using this sample and can add a check and do a pull, it would be great.

**Comment by AugustZellmer** [comment]
I know this isn't a complex change, but it's still not a change I've had time to make.
Edit: I will not have time to make this change.
